api/post.go

```diff
@@ -619,16 +619,23 @@ func deletePost(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	cchan := Srv.Store.Channel().CheckPermissionsTo(c.Session.TeamId, channelId, c.Session.UserId)
 	pchan := Srv.Store.Post().Get(postId)
+	uchan := Srv.Store.User().Get(c.Session.UserId)
 
-	if !c.HasPermissionsToChannel(cchan, "deletePost") {
+	if uresult := <-uchan; uresult.Err != nil {
+		c.Err = uresult.Err
 		return
-	}
-
-	if result := <-pchan; result.Err != nil {
-		c.Err = result.Err
+	} else if presult := <-pchan; presult.Err != nil {
+		c.Err = presult.Err
 		return
 	} else {
-		post := result.Data.(*model.PostList).Posts[postId]
+
+		user := uresult.Data.(*model.User)
+
+		if !c.HasPermissionsToChannel(cchan, "deletePost") && !strings.Contains(user.Roles,"admin"){
```

Comment:
Also I think you could be an admin of a diff team or even a private channel.  I think you still need to check HasPermissionsToChannel (or some variation) vs skipping when admin role checked.